### ISSUE TITLE
Reading a pcap fails if one packet has an orig_len > snap_len

### ISSUE THREAD

**Issue by Caelmyn** [courvoif/pcap-file#36]
Hi,

As said in the title, if a packet has a orig_len > snap_len, the crate returns an error: `Invalid field value: PacketHeader orig_len > snap_len`

However, such a pcap can be read by many softwares such as Wireshark, tcpdump, tcpreplay and Napatech's NIC drivers APIs.

Example of pcaps with such packets: [pcaps_err.zip](https://github.com/courvoif/pcap-file/files/13754049/pcaps_err.zip)

Also, here's a small code snippet you can c/p to reproduce the problem:

```rust
use std::fs::File;

use pcap_file::pcap::*;

/* ---------- */

const PCAP: &str = "dns.pcap";

/* ---------- */

fn main() {
    let file = File::open(PCAP).expect("failed to open the pcap");
    let mut reader = PcapReader::new(file).expect("failed to create a pcap reader");
    let mut count = 0usize;

    while let Some(pkt) = reader.next_packet() {
        let pkt = pkt.unwrap_or_else(|err| panic!("failed to read the pkt #{count}: {err}"));
        println!("pkt {count}: {:?}", pkt.data.as_ref());
        count += 1;
    }
}
```

**Comment by courvoif** [comment]
Hi, seems fair, I will remove the check.

**Comment by courvoif** [comment]
Fixed in #38
